### ISSUE TITLE
Sensors switch to NaN from time to time

### ISSUE THREAD

**Issue by cromefire** [EvanMulawski/FanControl.CorsairLink#120]
![image](https://github.com/EvanMulawski/FanControl.CorsairLink/assets/26320625/de7d72dc-d796-4857-aadc-3949cada6913)
From time to time the Hardware sensors seem to report NaN values, maybe there's something off with the polling requency or so? It doesn't seem like anything is being logged, at least no logfile is being created.

Maybe related, the fans just go to 100% from time to time as well for no reason, less frequently, but for longer.

**Comment by cromefire** [comment]
Okay just double checked in iCue and is just seems my Commander Pro was(/is?) wonkey, after reconnecting power and removing a USB device from it, it seems to be stable now (at least in iCue).

**Comment by cromefire** [comment]
So I've tried everything I could over the last few years basically, bit it seems to be a hardware issue with some AMD AM 4 motherboards' USB controller and the Commander Pro which can't be fixed, not even by an internal USB hub. So what would be pretty great is if the plugin would just keep reporting back the last value if a read isn't successful. Not forever of course, but it'd be great if it was able to do it for a few seconds at least as that temporary USB dropouts/bad values don't cause the FANs to spike.

I've tried catching the NaN using custom sensors and making it better that way, but no matter what I do there's just no good way in FanControl to get it to no immediately get all Fans at 100%. Once a value that's gone to NaN triggers a change in some sensors, it seems to just lose all concept of a hysteresis and ramps them up instantly (a concept which it regains when the values get back in line, so it ramps up in 1 second and then takes like 5-10 seconds to get back down...). I know it's not the greatest solution, but as the hardware can't be fixed, having the software be a bit more resilient to small glitches would be great. It always just seems to be 1-2 values that are missing.

Might make a PR for this this week if I can find some time.
